Formula/nu.rb

```diff
@@ -15,9 +15,45 @@ class Nu < Formula
 
   depends_on "pcre"
 
+  uses_from_macos "llvm" => [:build, :test]
+  uses_from_macos "swift" => :build # For libdispatch on Linux.
+
+  on_linux do
+    depends_on "gnustep-make" => :build
+    depends_on "gnustep-base"
+    depends_on "libffi"
+    depends_on "libobjc2"
+  end
+
+  # Clang must be used on Linux because GCC Objective-C support is insufficient.
+  fails_with :gcc
+
   def install
     ENV.delete("SDKROOT") if MacOS.version < :sierra
     ENV["PREFIX"] = prefix
+    # Some gnustep-make scripts located in libexecA.
+    ENV.append_path "PATH", Formula["gnustep-make"].libexec unless OS.mac?
+
+    unless OS.mac?
+      # Help linker find libdispatch from swift on Linux.
+      # This is only used for the mininush temporary compiler and is not needed for nush.
+      ldflags = %W[
+        "-L#{Formula["swift"].libexec}/lib/swift/linux"
+        "-Wl,-rpath,#{Formula["swift"].libexec}/lib/swift/linux"
+      ]
+      inreplace "Makefile", "LDFLAGS = ", "LDFLAGS = #{ldflags.join(" ")}"
+      # Don't hard code path to clang.
+      inreplace "tools/nuke", "/usr/bin/clang", ENV.cc
+
+      # Remove unused prefix from ffi.h
+      inreplace "objc/NuBridge.h", "x86_64-linux-gnu/", ""
+      inreplace "objc/NuBridge.m", "x86_64-linux-gnu/", ""
+      inreplace "objc/Nu.m", "x86_64-linux-gnu/", ""
+
+      # Nu does not (yet) support the GNUstep 2.0 Objective C runtime so it must use 1.9.
+      # If this flag is not changed from its default, the linker fails to find some symbols.
+      inreplace "Nukefile", "-fgnu-runtime", "-fobjc-runtime=gnustep-1.9"
```

Comment:
Hard-coding the `1.9` here seems really prone to bitrot.

Reply:
As documented here: https://github.com/gnustep/libobjc2, this is a defined standard whose version won't change over time.  We should check with new releases if the 2.0 run time is supported.  Maybe I should add this to comment?

Reply:
Yes. Along with links to documentation, and how to tell when `nu` supports the newer runtime.

Reply:
I have an interesting update to this.  The problem was the the Nukefile was also setting `-DGNU_RUNTIME=1`, which will result in the compiler trying to find symbols for the GNU runtime.  However if I remove `-DGNU_RUNTIME=1` and `-fgnu-runtime` from the CFLAGS, it automatically picks the the 1.9 run time without having to explicitly specify it.  

I believe this is essentially aligning it with what happens on macOS and should avoid bitrot as we're just removing some irrelevant flags.  If they decide to move to the Objective-C 2.0 runtime, upstream will have to remove these flags on Linux and the workaround will not be needed anymore.  I will update the comment to explain this.